File: src/python/core/domain/entities/general_info.py
from dataclasses import dataclass
from typing import Optional, List
from dataclasses_json import dataclass_json

from ..enums.seniority_level import SeniorityLevel as SeniorityLevelEnum
from .language import Language


@dataclass_json
@dataclass
class GeneralInfo:
    CandidateId: Optional[str] = None
    Fullname: Optional[str] = None
    TitleDetected: Optional[str] = None
    TitlePredicted: Optional[str] = None
    SeniorityLevel: Optional[SeniorityLevelEnum] = None
    YearsExperience: Optional[float] = None
    RelevantYears: Optional[float] = None
    IndustryMatch: Optional[str] = None
    TrajectoryPattern: Optional[str] = None
    MainIndustry: Optional[str] = None
    EnglishLevel: Optional[str] = None
    OtherLanguages: Optional[List[Language]] = None
    Location: Optional[str] = None

File: src/dotnet/Core/Domain/Entities/GeneralInfo.cs
using Rag.Candidates.Core.Domain.Enums;

namespace Rag.Candidates.Core.Domain.Entities;

public class GeneralInfo
{
    public string? CandidateId { get; set; }
    public string? Fullname { get; set; }
    public string? TitleDetected { get; set; }
    public string? TitlePredicted { get; set; }
    public SeniorityLevel? SeniorityLevel { get; set; }
    public int? YearsExperience { get; set; }
    public int? RelevantYears { get; set; }
    public string? IndustryMatch { get; set; }
    public string? TrajectoryPattern { get; set; }
    public string? MainIndustry { get; set; }
    public string? EnglishLevel { get; set; }
    public string[]? OtherLanguages { get; set; }
    public string? Location { get; set; }
    public string? RemoteWork { get; set; }
    public string? Availability { get; set; }
    public string? SalaryExpectations { get; set; }
    public string? NoticePeriod { get; set; }
}